tests/tree-status/src/main.rs

```diff
@@ -0,0 +1,820 @@
+    #[error("failed to decode Transaction")]
+    Transaction,
+    #[error("failed to decode instruction data: {0}")]
+    Instruction(#[from] bs58::decode::Error),
+    #[error("failed to parse pubkey: {0}")]
+    Pubkey(#[from] ParsePubkeyError),
+}
+
+#[derive(Debug, FromQueryResult, Clone)]
+struct MaxSeqItem {
+    max_seq: i64,
+    cnt_seq: i64,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, FromQueryResult, Clone)]
+struct MissingSeq {
+    missing_seq: i64,
+}
+
+#[derive(Debug, FromQueryResult)]
+struct AssetMaxSeq {
+    leaf_idx: i64,
+    seq: i64,
+}
+
+#[derive(Debug)]
+struct LeafNode {
+    leaf: Vec<u8>,
+    index: i64,
+}
+
+type MaybeLeafNode = Option<LeafNode>;
+
+#[derive(Parser)]
+#[command(next_line_help = true, author, version, about)]
+struct Args {
+    /// Solana RPC endpoint.
+    #[arg(long, short, alias = "rpc-url")]
+    rpc: String,
+
+    /// Number of concurrent requests for fetching transactions.
+    #[arg(long, default_value_t = 25)]
+    concurrency_tx: usize,
+
+    /// Number of concurrent processed trees.
+    #[arg(long, default_value_t = 5)]
+    concurrency_tree: usize,
+
+    /// Size of signatures queue
+    #[arg(long, default_value_t = 25_000)]
+    signatures_history_queue: usize,
+
+    /// Maximum number of retries for transaction fetching.
+    #[arg(long, short, default_value_t = 3)]
+    max_retries: u8,
+
+    /// Path to prometheus output
+    #[arg(long)]
+    prom: Option<String>,
+
+    // Optional prometheus group
+    #[arg(long)]
+    prom_group: Option<String>,
+
+    /// Prometheus metrics file update interval
+    #[arg(long, default_value_t = 1_000)]
+    prom_save_interval: u64,
+
+    #[command(subcommand)]
+    action: Action,
+}
+
+impl Args {
+    async fn get_pg_conn(&self) -> anyhow::Result<DatabaseConnection> {
+        match &self.action {
+            Action::CheckTree { pg_url, .. }
+            | Action::CheckTrees { pg_url, .. }
+            | Action::CheckTreeLeafs { pg_url, .. }
+            | Action::CheckTreesLeafs { pg_url, .. } => {
+                let options: PgConnectOptions = pg_url.parse().unwrap();
+
+                // Create postgres pool
+                let pool = PgPoolOptions::new()
+                    .min_connections(2)
+                    .max_connections(10)
+                    .connect_with(options)
+                    .await?;
+
+                // Create new postgres connection
+                Ok(SqlxPostgresConnector::from_sqlx_postgres_pool(pool))
+            }
+            Action::ShowTree { .. } | Action::ShowTrees { .. } => {
+                anyhow::bail!("show-tree and show-tress do not have connection to database")
+            }
+        }
+    }
+}
+
+#[derive(Subcommand, Clone)]
+enum Action {
+    /// Checks a single merkle tree to check if it's fully indexed
+    CheckTree {
+        #[arg(short, long)]
+        pg_url: String,
+        #[arg(short, long, help = "Tree pubkey")]
+        tree: String,
+    },
```

Comment:
Maybe its fine because all these action args tell you what they are for, but it was really nice in the bgtask main to have a header kind of saying overall what the tool was for and how it was used in practice.

So a minor request would be to add a similar header to tree-status and potentially the others.  Although this is just a "nice to have" and not a blocker by any means, its already much cleaned up overall.